### PR TITLE
Add integration tests, replace unwrap with error propagation, fix devenv scripts

### PR DESCRIPTION
**Pull request by doot**
## Summary

- **Tests**: Added 10 integration tests using a shared `OnceLock<Mutex<mockito::Server>>` singleton (port 4000) with `#[serial]` isolation. Covers happy path, edge cases (empty calendars, events without `dateTime`, multiple overlapping shows), error scenarios (TM 500, malformed JSON, garbage OIC response), and response header correctness. Mock handles are bound and `.assert()`-ed to confirm both upstreams are actually hit.
- **Error handling**: Replaced all `unwrap()` calls in `annotated_calendar` with structured `Result` propagation. `get_annotated` and `get_oic_events_cached` now return `Result<_>`. Used `tokio::join!` (not sequential `.await`) so a failing OIC fetch doesn't detach the TM task. Used `i64::try_from` for `overlap_window_hours` conversion so misconfiguration fails loudly. Used `unwrap_or_default()` for optional summary field.
- **Bug fix**: Corrected MIME type from `text/Calendar` to `text/calendar` (IANA-correct).
- **Settings**: Added configurable `overlap_window_hours` field (`u64`, default `3`) to `Settings`.
- **Cargo profiles**: Added `dev-debugger` profile (LLVM backend, full debug info, no optimization) for use when the debugger needs variable inspection. Removed bogus `inherits` key from `[profile.dev-debugger.package."*"]` (only valid on top-level profile tables).
- **devenv**: Bumped lockfile. Fixed container build scripts — added `-q` to suppress devenv progress output and `-r` to strip JSON quotes from `jq` output so the store path resolves correctly. Added `jq` and `lldb` to dev packages. Added Claude Code MCP server config.

## Test plan

- [x] `devenv test` passes locally (fmt + build + clippy + 10 cargo tests)
- [x] CI green: all 4 checks pass on `e3150f0`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/requests/cal.rs [hunk 1]
```diff
@@ -287,16 +308,404 @@ END:VEVENT
...
+                "/discovery/v2/events.json?venueId=abc&size=15&sort=date,asc&apikey=test_key",
+            )
+            .with_status(500)
+            .with_body("Internal Server Error")
+            .create();
+        server
+            .mock("GET", "/team-cal.php?team=123&tlev=0&tseq=0&season=456&format=iCal")
+            .with_body(TEST_CAL_DATA)
+            .create();
+
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs]: Fixed in e3150f0 — mock handles are now bound and .assert() is called after each request to verify both upstreams are hit.

File: tests/requests/cal.rs [hunk 2]
```diff
@@ -287,16 +308,404 @@ END:VEVENT
...
+                "GET",
+                "/discovery/v2/events.json?venueId=abc&size=15&sort=date,asc&apikey=test_key",
+            )
+            .with_body(TEST_EVENT_DATA)
+            .create();
+        server
+            .mock("GET", "/team-cal.php?team=123&tlev=0&tseq=0&season=456&format=iCal")
+            .with_body("<html>404 Not Found</html>")
+            .create();
+
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs]: Fixed in e3150f0 — mock handles are now bound and .assert() is called after each request to verify both upstreams are hit.

File: tests/requests/cal.rs [hunk 3]
```diff
@@ -287,16 +308,404 @@ END:VEVENT
...
+                "GET",
+                "/discovery/v2/events.json?venueId=abc&size=15&sort=date,asc&apikey=test_key",
+            )
+            .with_body("{not valid json at all")
+            .create();
+        server
+            .mock("GET", "/team-cal.php?team=123&tlev=0&tseq=0&season=456&format=iCal")
+            .with_body(TEST_CAL_DATA)
+            .create();
+
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs]: Fixed in e3150f0 — mock handles are now bound and .assert() is called after each request to verify both upstreams are hit.

File: tests/requests/cal.rs [hunk 4]
```diff
@@ -287,16 +308,404 @@ END:VEVENT
...
+        let mut server = mock_server().lock().unwrap();
+
+        server
+            .mock(
+                "GET",
+                "/discovery/v2/events.json?venueId=abc&size=15&sort=date,asc&apikey=test_key",
+            )
+            .with_status(500)
+            .with_body("Internal Server Error")
+            .create();
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs:628]: Fixed in e3150f0 — mock handles are now bound and .assert() is called after each request to verify both upstreams are hit.

File: src/annotated_calendar/mod.rs [hunk 5]
```diff
@@ -183,8 +174,8 @@ impl AnnotatedCalendar {
         for fe in &fox_events._embedded.events {
             if let Some(st) = fe.dates.start.date_time {
                 venue_event_infos.push(VenueEventInfo {
-                    lower_bound: st - Duration::hours(3),
-                    upper_bound: st + Duration::hours(3),
+                    lower_bound: st - Duration::hours(overlap_window_hours as i64),
+                    upper_bound: st + Duration::hours(overlap_window_hours as i64),
```
[the remark this answers is not in the document]
Comment (doot) [reply src/annotated_calendar/mod.rs]: Fixed in e3150f0 via i64::try_from(overlap_window_hours).map_err(Error::wrap)? which propagates an error for values > i64::MAX rather than silently wrapping to a negative duration.

File: src/annotated_calendar/mod.rs [hunk 6]
```diff
@@ -65,8 +65,8 @@ impl AnnotatedCalendar {
         });
 
         // Block until both responses finish
-        let mut oic_calendar = oic_handle.await.unwrap();
-        let tm_venue_events = tm_handle.await.unwrap();
+        let mut oic_calendar = oic_handle.await.map_err(Error::wrap)??;
+        let tm_venue_events = tm_handle.await.map_err(Error::wrap)??;
```
[the remark this answers is not in the document]
Comment (doot) [reply src/annotated_calendar/mod.rs]: Fixed in e3150f0 — replaced sequential .await with tokio::join!(oic_handle, tm_handle) so both tasks always run to completion before any error is propagated.

File: Cargo.toml [hunk 7]
```diff
@@ -52,10 +52,23 @@ pretty_assertions = "1.4.1"
...
+[profile.dev-debugger]
+# Alternative dev profile that disables dev profile compile time optimizations for use in a debugger.
+inherits = "dev"
+debug = true
+opt-level = 0
+codegen-backend = "llvm"
+
+[profile.dev-debugger.package."*"]
+# Alternative dev profile for dependencies that disables dev profile compile time optimizations for use in a debugger.
+inherits = "dev"
```
[the remark this answers is not in the document]
Comment (doot) [reply Cargo.toml]: Fixed in e3150f0 — removed the inherits key from [profile.dev-debugger.package.*] (inherits is only valid on top-level profile tables).

File: devenv.nix [hunk 8]
```diff
@@ -74,7 +93,7 @@ in {
 
         echo "Building docker image and copying it to local docker daemon..."
 
-        copyscript=$(devenv build outputs.prod_image_copy_local)
+        copyscript=$(devenv build outputs.prod_image_copy_local | jq '.["outputs.prod_image_copy_local"]')
```
[the remark this answers is not in the document]
Comment (doot) [reply devenv.nix]: Fixed in e3150f0 — both container tasks now use devenv build -q | jq -r so devenv's progress output no longer corrupts jq's input and the path is unquoted.

File: devenv.nix [hunk 9]
```diff
@@ -97,7 +116,7 @@ in {
 
         echo "Building docker image and copying it to remote registry..."
 
-        copyscript=$(devenv build outputs.prod_image_copy_registry)
+        copyscript=$(devenv build outputs.prod_image_copy_registry | jq '.["outputs.prod_image_copy_registry"]')
```
[the remark this answers is not in the document]
Comment (doot) [reply devenv.nix]: Fixed in e3150f0 — both container tasks now use devenv build -q | jq -r so devenv's progress output no longer corrupts jq's input and the path is unquoted.

File: src/common/settings.rs [hunk 10]
```diff
@@ -9,6 +9,12 @@ pub struct Settings {
     pub tm_page_size: u64,
     pub oic_cal_base_url: String,
     pub cache_duration: u64,
+    #[serde(default = "default_overlap_window_hours")]
+    pub overlap_window_hours: i64,
+}
+
+fn default_overlap_window_hours() -> i64 {
+    3
```
[the remark this answers is not in the document]
Comment (doot) [reply src/common/settings.rs:17]: Addressed in e3150f0 via i64::try_from at the call site. The struct field stays u64 to prevent negative YAML values; the cast site now propagates an error instead of silently wrapping.

File: tests/requests/cal.rs [hunk 11]
```diff
@@ -287,16 +308,404 @@ END:VEVENT
...
+        let oic_mock = server
+            .mock("GET", "/team-cal.php?team=123&tlev=0&tseq=0&season=456&format=iCal")
+            .with_body(TEST_CAL_DATA)
+            .create();
+
+        drop(server);
+        let resp = request.get("/api/cal/123/456").await;
+
+        assert_eq!(resp.status_code(), 200);
+        assert_eq!(resp.header("content-type"), "text/Calendar; charset=utf-8");
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs]: Fixed in 31833c3 — both the controller and the test assert text/calendar (lowercase) per IANA spec.

File: devenv.nix [hunk 12]
```diff
@@ -74,7 +74,7 @@ in {
 
         echo "Building docker image and copying it to local docker daemon..."
 
-        copyscript=$(devenv build outputs.prod_image_copy_local)
+        copyscript=$(devenv build outputs.prod_image_copy_local | jq '.["outputs.prod_image_copy_local"])
```
[the remark this answers is not in the document]
Comment (doot) [reply devenv.nix]: Stale — the jq invocation was rewritten in this PR and is now properly balanced.

File: devenv.nix [hunk 13]
```diff
@@ -137,10 +137,11 @@ in {
 
   # Hacky way to avoid putting CLI in 'packages' and delay building the CLI until needed
   scripts."${project_name}-cli".exec = ''
-    $(devenv build -q outputs.oic_fox_fuckery_cli)/bin/${project_name}-cli "$@"
+    $(devenv build -q outputs.oic_fox_fuckery_cli | jq '.["outputs.oic_fox_fuckery_cli"])/bin/${project_name}-cli "$@"
```
[the remark this answers is not in the document]
Comment (doot) [reply devenv.nix]: Stale — devenv.nix:159 (the CLI script) already uses -r and balanced quotes. The container tasks were separately fixed in e3150f0.

File: devenv.nix [hunk 14]
```diff
@@ -97,7 +97,7 @@ in {
 
         echo "Building docker image and copying it to remote registry..."
 
-        copyscript=$(devenv build outputs.prod_image_copy_registry)
+        copyscript=$(devenv build outputs.prod_image_copy_registry | jq '.["outputs.prod_image_copy_registry"]')
```
[the remark this answers is not in the document]
Comment (doot) [reply devenv.nix:120]: Already present — pkgs.jq is added at devenv.nix:82 for non-container/non-test builds.

File: devenv.nix [hunk 15]
```diff
@@ -65,6 +65,7 @@ in {
     ]
     ++ lib.optionals config.container.isBuilding [
       oic_fox_fuckery_cli # Project package
+      pkgs.jq # Needed for build scripts that manipulate devenv output
     ];
```
[the remark this answers is not in the document]
Comment (doot) [reply devenv.nix:87]: Already present — pkgs.jq is added at devenv.nix:82 for non-container/non-test builds.

File: src/annotated_calendar/mod.rs [hunk 16]
```diff
@@ -65,8 +65,12 @@ impl AnnotatedCalendar {
...
 
         // Block until both responses finish
-        let mut oic_calendar = oic_handle.await.unwrap();
-        let tm_venue_events = tm_handle.await.unwrap();
+        let mut oic_calendar = oic_handle
+            .await
+            .map_err(|e| Error::wrap(loco_rs::Error::string(&e.to_string())))??;
+        let tm_venue_events = tm_handle
+            .await
+            .map_err(|e| Error::wrap(loco_rs::Error::string(&e.to_string())))??;
```
[the remark this answers is not in the document]
Comment (doot) [reply src/annotated_calendar/mod.rs]: Already simplified in this PR to .map_err(Error::wrap)?? — the double-wrap is gone.

File: tests/requests/cal.rs [hunk 17]
```diff
@@ -1,3 +1,9 @@
+// All tests are #[serial] — the MutexGuard is explicitly dropped before .await,
+// but clippy can't track drop() calls for this lint.
+#![allow(clippy::await_holding_lock)]
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs:3]: The allow is crate-level to avoid per-function boilerplate across ten tests. The pattern (explicit drop(server) before .await) is correct — clippy cannot track drop() calls for this lint. Leaving as-is.

File: tests/requests/cal.rs [hunk 18]
```diff
@@ -9,6 +15,21 @@ use pretty_assertions::assert_eq;
...
+#[cfg(test)]
+fn mock_server() -> &'static Mutex<mockito::Server> {
+    static SERVER: OnceLock<Mutex<mockito::Server>> = OnceLock::new();
+    SERVER.get_or_init(|| {
+        Mutex::new(mockito::Server::new_with_opts(mockito::ServerOpts {
+            host: "0.0.0.0",
+            port: 4000,
+            ..Default::default()
+        }))
+    })
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs]: The doc comment describes mockito's RAII cleanup (Mock drop removes the registered expectation), not a call to Server::reset(). Calling reset() inside mock_server() would clear mocks on every lock acquisition, including mid-test. The pattern is correct as-is.

File: tests/requests/cal.rs [hunk 19]
```diff
@@ -287,16 +308,404 @@ END:VEVENT
...
+    request::<App, _, _>(|request, _ctx| async move {
+        let mut server = mock_server().lock().unwrap();
+
+        server
+            .mock(
+                "GET",
+                "/discovery/v2/events.json?venueId=abc&size=15&sort=date,asc&apikey=test_key",
+            )
+            .with_status(500)
+            .with_body("Internal Server Error")
```
[the remark this answers is not in the document]
Comment (doot) [reply tests/requests/cal.rs:627]: The plain-text body is intentional — it causes a JSON deserialization error, which exercises the error propagation path. The handler does not currently call error_for_status(), so a valid-JSON body with status 500 would be parsed successfully and return 200. Testing status-code rejection separately would require a handler change, which is out of scope for this PR.

File: devenv.nix [hunk 20]
```diff
@@ -97,7 +98,7 @@ in {
 
         echo "Building docker image and copying it to remote registry..."
 
-        copyscript=$(devenv build outputs.prod_image_copy_registry)
+        copyscript=$(devenv build outputs.prod_image_copy_registry | jq '.["outputs.prod_image_copy_registry"]')
 
         echo "Pushing image to registry via $copyscript..."
         $copyscript/bin/copy-to-registry --dest-creds ${registry_user}:$REGISTRY_API_KEY
```
[the remark this answers is not in the document]
Comment (doot) [reply devenv.nix:122]: Acknowledged — this is a dev-only push script (not run in CI or production). Secure secret handling for registry pushes is out of scope for this PR.
